components/mosq/port/callbacks.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Roger Light <roger@atchoo.org>
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * SPDX-FileContributor: 2024 Espressif Systems (Shanghai) CO LTD
+ */
+#include "mosquitto_internal.h"
+#include "mosquitto_broker.h"
+#include "memory_mosq.h"
+#include "mqtt_protocol.h"
+#include "send_mosq.h"
+#include "util_mosq.h"
+#include "utlist.h"
+#include "lib_load.h"
+
+
+int mosquitto_callback_register(
```

Comment:
Will we make it more flexible to use the callbacks in the future?

Reply:
we can extend it in future, but this version doesn't support plugins

Reply:
I will not support plugins in the near future, but I'd like to implement callbacks in user configs, which would be useful to monitor whenever 
* client connects/disconnects
* provide a custom means to reject connection (based on IP, client id, user, password....)
* message receive -- with client id, topic, payload.

(experimented with that in https://github.com/david-cermak/esp-protocols/commit/d24019a133c959d56a02e2196fc20fa9c5b81957)

but that's still won't happen in the first published version